src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapperUtil.java

```diff
@@ -74,6 +76,34 @@ public static void validateByteVectorValue(float value) {
         }
     }
 
+    /**
+     * Validate if the given byte vector is supported by the given space type
+     *
+     * @param vector     the given vector
+     * @param spaceType  the given space type
+     */
+    public static void validateByteVector(byte[] vector, SpaceType spaceType) {
+        if (spaceType == SpaceType.COSINESIMIL && isZeroVector(vector)) {
+            throw new IllegalArgumentException(
+                String.format(Locale.ROOT, "zero vector is not supported when space type is [%s]", spaceType.getValue())
+            );
+        }
+    }
+
+    /**
+     * Validate if the given float vector is supported by the given space type
+     *
+     * @param vector     the given vector
+     * @param spaceType  the given space type
+     */
+    public static void validateFloatVector(float[] vector, SpaceType spaceType) {
```

Comment:
should we make these parts of SpaceType enum class? Also, does these validations true for all engines?

Reply:
the zero vector validation is same for all engines, but maybe not true for other validations introduced in the future. For example, the `dot product` metric in Lucene requires all vectors must be normalized, but Faiss doesn't, so I tend to keep these part in this class, any thoughts?

Reply:
if its true for all engines that it makes more sense to add as part of enum class.